internal/charmstore/migrations.go

```diff
@@ -350,6 +357,77 @@ func ncmUpdateBaseEntity(db StoreDatabase, baseEntity *preNCMBaseEntity) error {
 	return nil
 }
 
+func migrateToArchiveDownloadStatsOnly(db StoreDatabase) error {
+	entities := db.Entities()
+	// Find entities with the highest revision where extra-info is set
+	// then recreate a structure that can be used on mongodoc.Entity
+	iter := entities.Pipe(
+		[]bson.D{
+			{{"$match", bson.D{{"extrainfo.legacy-download-stats", bson.D{{"$exists", true}}}}}},
+			{{"$sort", bson.D{{"revision", 1}}}},
+			{{"$group", bson.D{
+				{"_id", "$baseurl"},
+				{"url", bson.D{{"$last", "$_id"}}},
+				{"promulgated-url", bson.D{{"$last", "$promulgated-url"}}},
+				{"revision", bson.D{{"$max", "revision"}}},
+				{"legacy-download-stats", bson.D{{"$last", "$extrainfo.legacy-download-stats"}}},
+			}}},
+			{{"$project", bson.D{
+				{"_id", "$url"},
+				{"promulgated-url", "$promulgated-url"},
+				{"extrainfo.legacy-download-stats", "$legacy-download-stats"},
+			}}},
+		}).Iter()
+
+	var stats stats
+	var entity mongodoc.Entity
+	for iter.Next(&entity) {
+		legacyDownloadStats, ok := entity.ExtraInfo[params.LegacyDownloadStats]
+		if !ok {
+			return errgo.Newf("cannot retrieve the legacy download stats on entity: %v", entity.URL)
+		}
+		if len(legacyDownloadStats) == 0 {
+			continue
+		}
+		key, err := stats.key(db, EntityStatsKey(entity.URL, params.StatsArchiveDownload), true)
+		if err != nil {
+			return errgo.Mask(err)
+		}
+		err = setArchiveDownloadStats(db, key, legacyDownloadStats)
+		if err != nil {
+			return errgo.Notef(err, "cannot set the download count on entity: %v", entity.URL)
+		}
+		if entity.PromulgatedURL != nil {
+			key, err := stats.key(db, EntityStatsKey(entity.PromulgatedURL, params.StatsArchiveDownloadPromulgated), true)
+			if err != nil {
+				return errgo.Mask(err)
+			}
+			err = setArchiveDownloadStats(db, key, legacyDownloadStats)
+			if err != nil {
+				return errgo.Notef(err, "cannot set the download count on entity: %v", entity.URL)
+			}
+		}
+	}
+	if err := iter.Err(); err != nil {
+		return errgo.Notef(err, "cannot iterate through entities")
+	}
+	return nil
+}
+
+func setArchiveDownloadStats(db StoreDatabase, key string, legacyDownloadStats []byte) error {
```

Comment:
Well, technically, I think @jrwren is correct - if there were already any downloads
for that specific date, this will add to them not set them, right?